packages/sanity/src/tasks/src/tasks/components/form/tasksFormBuilder/TasksNotificationTarget.tsx

```diff
@@ -0,0 +1,80 @@
+import {isImageSource} from '@sanity/asset-utils'
+import imageUrlBuilder from '@sanity/image-url'
+import {useEffect, useMemo} from 'react'
+import deepEquals from 'react-fast-compare'
+import {
+  DEFAULT_STUDIO_CLIENT_OPTIONS,
+  type ObjectFieldProps,
+  set,
+  useClient,
+  useFormValue,
+  useWorkspace,
+} from 'sanity'
+
+import {useDocumentPreviewValues} from '../../../hooks/useDocumentPreviewValues'
+import {type TaskContext, type TaskDocument} from '../../../types'
+import {CurrentWorkspaceProvider} from '../CurrentWorkspaceProvider'
+
+function TasksNotificationTargetInner(props: ObjectFieldProps<TaskDocument>) {
+  const {inputProps} = props
+  const {onChange} = inputProps
+  const {target, _id, context} = useFormValue([]) as TaskDocument
+  const {title: workspaceTitle, basePath} = useWorkspace()
+  const client = useClient(DEFAULT_STUDIO_CLIENT_OPTIONS)
+  const imageBuilder = useMemo(() => imageUrlBuilder(client), [client])
+  const documentId = target?.document?._ref ?? ''
+  const documentType = target?.documentType ?? ''
+
+  const {isLoading: previewValuesLoading, value} = useDocumentPreviewValues({
+    documentId,
+    documentType,
+  })
+  const targetContentTitle = value?.title || null
+  const imageUrl = isImageSource(value?.media)
+    ? imageBuilder.image(value.media).width(96).height(96).url()
+    : null
+
+  useEffect(() => {
+    if (documentId && documentType && previewValuesLoading) {
+      // Wait until the preview values are loaded
+      return
+    }
+
+    const studioUrl = new URL(`${window.location.origin}${basePath}/`)
+    studioUrl.searchParams.set('sidebar', 'tasks')
+    studioUrl.searchParams.set('selectedTask', _id)
+    studioUrl.searchParams.set('viewMode', 'edit')
+
+    const notificationTarget: TaskContext['notification'] = {
+      url: studioUrl.toString(),
+      workspaceTitle,
+      targetContentImageUrl: imageUrl,
+      targetContentTitle: targetContentTitle,
+    }
+    if (deepEquals(notificationTarget, context)) return
```

Comment:
I believe this comparison will always yield `false` because it compares the next notification object (`notificationTarget`) with `context`, which has the current notification object. I think the correct check should be:

```ts
 if(isEqual(notificationTarget, context?.notification)) return
```

That is, to use `context?.notification` instead of just `context` when comparing.

Reply:
You are right